[KOGITO-8818] Rearrange guides to add operator and quarkus cloud (#313)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/cloud/operator/workflow-status-conditions.adoc

= Understanding Workflow Services Status Conditions
:compat-mode!:
// Metadata:
:description: Description of the status and conditions of a workflow deployed by the operator
:keywords: kogito, workflow, serverless, operator, kubernetes, minikube, status, conditions

This document describes the status and conditions of the `KogitoServerlessWorkflow` object deployed by the {operator_name}.

link:https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties[Kubernetes Status] is an important property to observe in order to understand what is currently happening with the object. It can also help you troubleshoot or integrate with other objects in the cluster. 

== General Status

The table bellow lists the general structure of a workflow status.

> .Description of KogitoServerlessWorkflow resource object

Reviewer: ```suggestion
.Description of KogitoServerlessWorkflow status object
```

The following table lists the possible conditions.

.Conditions Scenarios in Development
[cols="0,0,1,2"]
|===
|Condition | Status | Reason | Description

| Built
| Unknown
| 
| In development profile there's no build process, so this state remains in `Unknown` status

| Running
| True
| 
| The workflow is running and in healthy state

| Running
| False
| WaitingForDeployment
| The workflow is waiting for the underlying deployment object to have a minimum availability

| Running
| False
| DeploymentFailure
| There was a problem with the underlying deployment object. Check the message in this condition and the workflow pod logs for more info

| Running
| False
| DeploymentIsUnavailable
| The underlying deployment object doesn't have the minimum availability for this workflow. Check the message in this condition and the workflow pod logs for more info

| Running
| False
| AttemptToRedeployFailed
| If the workflow deployment is not available, the operator will try to rollout the deployment three times before entering in this stage. Check the message in this condition and the workflow pod logs for more info

|===

> In normal conditions the workflow will transition from `Running`, `WaitingForDeployment` condition to `Running`. In case something wrong happens, consult the section xref:cloud/operator/developing-workflows.adoc#troubleshooting[Workflow Troubleshooting in Development].

Reviewer 2: ```suggestion
In normal conditions, the workflow will transition from `Running`, `WaitingForDeployment` condition to `Running`. In case something wrong happens, consult the section xref:cloud/operator/developing-workflows.adoc#troubleshooting[Workflow Troubleshooting in Development].
```

[cols="0,0,1,2"]
|===
|Condition | Status | Reason | Description

| Built
| False
| BuildIsRunningReason
| The build is currently running

| Built
| False
| BuildFailedReason
| The build has failed and any attempts to have it run failed

| Built
| True
| 
| Last build has finished successfully

| Running
| True
| 
| The workflow is running and in healthy state

| Running
| False
| WaitingForPlatform
| The workflow can't have a pod running until a `KogitoServerlessPlatform` is ready

| Running
| False
| WaitingForBuildReason
| The workflow is waiting for the build to finish to start running

| Running
| False
| WaitingForDeployment
| The workflow is waiting for the underlying deployment object to have a minimum availability

| Running
| False
| DeploymentFailure
| There was a problem with the underlying deployment object. Check the message in this condition and the workflow pod logs for more info

| Running
| False
| DeploymentIsUnavailable
| The underlying deployment object doesn't have the minimum availability for this workflow. Check the message in this condition and the workflow pod logs for more info

| Running
| False
| AttemptToRedeployFailed

> | If the workflow deployment is not available, the operator will try to rollout the deployment three times before entering in this stage. Check the message in this condition and the workflow pod logs for more info

Reviewer 2: ```suggestion
| If the workflow deployment is not available, the operator will try to roll out the deployment three times before entering this stage. Check the message in this condition and the workflow pod logs for more info
```

 
|===

> The normal conditions for the workflow resource is to place a `KogitoServerlessBuild` to run and wait for it to finish. As soon as the image is ready, the workflow transitions to the deployment phase, which is to provision a new workflow service pod to run with the built image.

Reviewer 2: ```suggestion
The normal conditions for the workflow resource are to place a `KogitoServerlessBuild` to run and wait for it to finish. As soon as the image is ready, the workflow transitions to the deployment phase, which is to provision a new workflow service pod to run with the built image.
```
